packages/dialtone-vue2/components/stack/utils.js

```diff
@@ -24,17 +32,36 @@ export function getDefaultDirectionClass (direction) {
     : null;
 }
 
-export function getResponsiveClasses (direction) {
+function getResposiveDirectionClasses (direction) {
   if (directionPropType(direction) === 'object') {
     return [
       ...DT_STACK_RESPONSIVE_BREAKPOINTS.map((breakpoint) => {
         return direction[breakpoint]
-          ? `d-stack--${breakpoint}--${direction[breakpoint]}`
+          ? `d-stack--${breakpoint}--${DT_STACK_DIRECTION[direction[breakpoint]]}`
           : null;
       })];
-  } else { return null; }
+  } else { return []; }
+}
+
+function getResposiveGapClasses (gap) {
+  if (directionPropType(gap) === 'object') {
+    return [
+      ...DT_STACK_RESPONSIVE_BREAKPOINTS.map((breakpoint) => {
+        return DT_STACK_GAP.includes(gap[breakpoint])
+          ? `d-stack--${breakpoint}--gap-${gap[breakpoint]}`
```

Comment:
based on BEM conventions we should never go over 1 level of nesting: https://getbem.com/faq/#css-nested-elements

Easy fix, just change to a single dash:
```
d-stack--sm-gap-500
```